addons/survey/models/survey_question.py

```diff
@@ -273,10 +277,136 @@ def _index(self):
+        """ Compute statistical data for questions by counting number of vote per choice on basis of filter """
+        all_questions_data = []
+        for question in self:
+            question_data = {'question': question, 'is_page': question.is_page}
+
+            if question.is_page:
+                all_questions_data.append(question_data)
+                continue
+
+            # fetch answer lines, separate comments from real answers
+            all_line_ids = user_input_lines.filtered(lambda line: line.question_id == question)
+            if question.question_type in ['simple_choice', 'multiple_choice', 'matrix']:
+                answer_line_ids = all_line_ids.filtered(
+                    lambda line: line.answer_type == 'suggestion' or (
+                        line.answer_type == 'char_box' and question.comment_count_as_answer)
+                    )
+                comment_line_ids = all_line_ids.filtered(lambda line: line.answer_type == 'char_box')
+            else:
+                answer_line_ids = all_line_ids
+                comment_line_ids = all_line_ids.browse([])
+            question_data.update(
+                all_line_ids=all_line_ids,
+                answer_line_ids=answer_line_ids,
+                answer_line_done_ids=answer_line_ids.filtered(lambda line: not line.skipped),
+                answer_input_done_ids=answer_line_ids.filtered(lambda line: not line.skipped).mapped('user_input_id'),
+                answer_input_skipped_ids=answer_line_ids.filtered(lambda line: line.skipped).mapped('user_input_id'),
+                comment_line_ids=comment_line_ids)
+            question_data.update(question._get_stats_summary_data(answer_line_ids))
+
+            # prepare table and graph data
+            table_data, graph_data = question._get_stats_data(answer_line_ids)
+            question_data['table_data'] = table_data
+            question_data['graph_data'] = json.dumps(graph_data)
+
+            all_questions_data.append(question_data)
+        return all_questions_data
+
+    def _get_stats_data(self, user_input_lines):
+        if self.question_type in ['simple_choice']:
+            return self._get_stats_data_answers(user_input_lines)
+        elif self.question_type in ['multiple_choice']:
+            table_data, graph_data = self._get_stats_data_answers(user_input_lines)
+            return table_data, [{'key': self.title, 'values': graph_data}]
+        elif self.question_type in ['matrix']:
+            return self._get_stats_graph_data_matrix(user_input_lines)
+        return [line for line in user_input_lines], []
+
+    def _get_stats_data_answers(self, user_input_lines):
+        """ [{"text": "Once a day", "count": 0, "answer_id": 1, "answer_score": 0.0}, ...] """
+        suggested_answers = [answer for answer in self.mapped('suggested_answer_ids')]
+        if self.comment_count_as_answer:
+            suggested_answers += [self.env['survey.question.answer']]
```

Comment:
Meh.

I wonder if you could create a temporary suggested.question.answer with `_('Other (see comments)')` as value to make this much clearer.
And then unlink it when you have your structure ready.

But maybe it's too much.

Reply:
In my opinion that solution is also a bit meh. Maybe it should be managed at model level (comments allowed would create an "other" question answer instead of managing it as a corner case in templates / controllers / statistics).